Formula/ffmpeg.rb

```diff
@@ -1,16 +1,44 @@
 class Ffmpeg < Formula
   desc "Play, record, convert, and stream audio and video"
   homepage "https://ffmpeg.org/"
-  url "https://ffmpeg.org/releases/ffmpeg-3.1.4.tar.bz2"
-  sha256 "7c99df75a4dc12d22c0f1ed11d0acf98cac1f8b5fe7a7434344b167f810bcbfa"
-  head "https://github.com/FFmpeg/FFmpeg.git"
+
+  stable do
+    url "https://ffmpeg.org/releases/ffmpeg-3.1.4.tar.bz2"
+    sha256 "7c99df75a4dc12d22c0f1ed11d0acf98cac1f8b5fe7a7434344b167f810bcbfa"
+
+    option "with-sdl", "Enable FFplay media player"
+    option "with-openh264", "Enable OpenH264 library"
+    deprecated_option "with-ffplay" => "with-sdl"
+
+    depends_on "sdl" => :optional
+    depends_on "nasm" => :build if build.with? "openh264"
+
+    # Remove when ffmpeg has support for openh264 1.6.0
+    # See https://github.com/cisco/openh264/issues/2505
+    # Master now has support, but not the 3.1.x branch
+    resource "openh264-1.5.0" do
+      url "https://github.com/cisco/openh264/archive/v1.5.0.tar.gz"
+      sha256 "98077bd5d113c183ce02b678733b0cada2cf36750370579534c4d70f0b6c27b5"
+    end
+  end
 
   bottle do
     sha256 "72394f281e42f49816f19a1fe842963ce710be992d9cdafb0b147782ea0a1af1" => :sierra
     sha256 "7c9200710f18df67ddb475436fc312ee4a6c2b0ab6f758c373c28873f84cc89b" => :el_capitan
     sha256 "cc0b73f7ba46f677e4e93e4c274c701895a28e14e97fa335dfea280285ad8df3" => :yosemite
   end
 
+  head do
+    url "https://github.com/FFmpeg/FFmpeg.git"
+
+    # Support for SDL1 has been removed from master
+    option "with-sdl2", "Enable FFplay media player"
```

Comment:
The `--with-sdl2` option is not as visible from the CLI as I might like, it seems that because of the way [info.rb](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cmd/info.rb#L40) parses args to allow for its three different command forms, passing `--HEAD` when running `brew info <formula>` currently has no effect on the display of `head` specific options, dependencies, requirements, etc. However, it is still shown when `brew options ffmpeg --HEAD` is run, and in the formula itself, so perhaps that is enough.

On a side note, I wonder if there is any merit in improving the `info` command's ability to take formula specific modifiers. It may be best to maintain `options` specialization there to avoid cumbersome code.... hmm...